lib/MusicBrainz/Server/EditSearch/Predicate/Voter.pm

```diff
@@ -43,32 +44,65 @@ sub valid {
 sub voter_clause {
     my $self = shift;
     my $sql = 'vote.editor ';
+    my @args;
 
     if ($self->operator eq 'subscribed') {
         $sql .= 'IN (
              SELECT subscribed_editor
                FROM editor_subscribe_editor
               WHERE editor = ?
         )';
+        push @args, $self->user->id;
     } elsif ($self->operator eq 'not_subscribed') {
         $sql .= 'NOT IN (
              SELECT subscribed_editor
                FROM editor_subscribe_editor
               WHERE editor = ?
         )';
-    } elsif ($self->operator eq '!=' || $self->operator eq 'not_me') {
+        push @args, $self->user->id;
+    } elsif ($self->operator eq '=') {
+        $sql .= '= ?';
+        push @args, $self->voter_id;
+    } elsif ($self->operator eq '!=') {
         $sql .= '!= ?';
-    } else {
+        push @args, $self->voter_id;
+    } elsif ($self->operator eq 'me') {
         $sql .= '= ?';
+        push @args, $self->user->id;
+    } elsif ($self->operator eq 'not_me') {
+        $sql .= '!= ?';
+        push @args, $self->user->id;
+    } elsif ($self->operator eq 'limited') {
+        # Please keep the beginner logic in sync with Report::LimitedEditors and Entity::Editor
```

Comment:
If the goal is to prevent the different parts of the code to get out of sync, a similar comment should probably be added in these other parts of the code.